projects/ngx-translate/core/src/lib/translate.service.ts

```diff
@@ -308,22 +315,13 @@ export class TranslateService {
         observables: boolean = false;
       for (let k of key) {
         result[k] = this.getParsedResult(translations, k, interpolateParams);
-        if (typeof result[k].subscribe === "function") {
+        if (isObservable(result[k])) {
           observables = true;
         }
       }
       if (observables) {
-        let mergedObs: Observable<string>;
-        for (let k of key) {
-          let obs = typeof result[k].subscribe === "function" ? result[k] : of(result[k] as string);
-          if (typeof mergedObs === "undefined") {
-            mergedObs = obs;
-          } else {
-            mergedObs = merge(mergedObs, obs);
-          }
-        }
-        return mergedObs.pipe(
-          toArray(),
```

Comment:
This code is just refactored because it was unnecessarily complicated. Functionality should be the same.